core/src/modules/tableSpeculation.ts

```diff
@@ -57,4 +59,17 @@ export namespace TableSpeculation {
 
     return databaseType;
   }
+
+  export function suggestKey(
+    key: string,
+    model: GrouparooModel,
+    existingProperties: Property[]
+  ): string {
+    for (const property of existingProperties) {
+      if (property.key === key) {
```

Comment:
Rather than doing a whole loop like this, you can pluck a matching property from the array:

```ts
export function suggestKey(
    key: string,
    model: GrouparooModel,
    existingProperties: Property[]
  ): {
  const matchingProperty = existingProperties.find(p => p.key === key);
  return matchingProperty ? `${ConfigWriter.generateId(model.name)}_${key}` : key;
}

The return type of the method will also be discovered automatically since we are always returning a string - no need to type the method's return up top
```